### PR TITLE
`glslangValidator`: Propagage error from `glslang::OutputSpv[Hex|Bin]`

### LINKED ISSUE #3218: glslangValidator exits successfully if it cannot write output file
glslangValidator should exit unsuccessfully if it fails to write the output file, so that it can be integrated into build systems reliably. Currently, it exits successfully:

```
$ cat shader.frag
#version 460
void main() {}
$ ls
shader.frag
$ glslangValidator -V -o non-existent/shader.frag.spv shader.frag
shader.frag
ERROR: Failed to open file: non-existent/shader.frag.spv
$ echo "$?"
0
```

### PR DESCRIPTION
**Pull request by allanmac**
Closes #3218.

Also it appears `KhronosGroup:main` acquired some trailing spaces from a previous commit.

### CODE REVIEW

**Review by arcady-lunarg — CHANGES_REQUESTED** [review]
Hi, could you separate out the whitespace fixes into a separate commit? Also, I would suggest rewording the commit description to more clearly state what it is doing. The fix itself is good though, thanks for the PR.

**Comment by allanmac** [comment]
> Hi, could you separate out the whitespace fixes into a separate commit? Also, I would suggest rewording the commit description to more clearly state what it is doing. The fix itself is good though, thanks for the PR.

#3232

**Review by arcady-lunarg — APPROVED** [review]
LGTM
